ts/services/SampleService.ts

```diff
@@ -0,0 +1,221 @@
+
+                const rows = [];
+                response.samples.forEach((s) => {
+                  rows.push([s.timestamp, s.rollup, s.data.cnt, s.data.min, s.data.avg, s.data.max, s.data.sum]);
+                });
+
+                const table = Table(header, rows, {
+                  align : 'left',
+                  borderStyle : 1,
+                  color : 'white',
+                  headerAlign : 'left'
+                });
+
+                logger.info(table.render());
+            }
+            if (config.format === 'csv') {
+                logger.info('timestamp, rollup, cnt, min, avg, max, sum, val');
+                const rows = [];
+                response.samples.forEach((s) => {
+                    let line = `${s.timestamp}, ${s.rollup}, `;
+                    line += `${s.data.cnt}, ${s.data.min}, ${s.data.avg}, ${s.data.max}, ${s.data.sum}, ${s.data.val}`;
+                    logger.info(line);
+                });
+            }
+
+            if (config.format === 'json') {
+                logger.info(JSON.stringify(response, null, 2));
+            }
+        } catch (e) {
+            logger.error('There was an error listing the elements: ' + e);
+        }
+    }
+
+    public buildSampleUri(elementId, metricId, config, logger) {
+        let queryUri = `${config.endpoint}`;
+        queryUri += `/ui/elements/${elementId}`;
+        queryUri += `/metrics/${metricId}`;
+        queryUri += `/samples?aggregations=ACTUAL`;
+        if (config.rollup) {
+            queryUri += `&rollup=${config.rollup}`;
+        }
+        if (config.duration) {
+            queryUri += `&duration=${config.duration}`;
+        }
+        if (config.startTime) {
+            queryUri += `&startTime=${config.startTime}`;
+        }
+        if (config.endTime) {
+            queryUri += `&endTime=${config.endTime}`;
+        }
+
+        return queryUri;
+    }
+
+    public async summarizeSamples(config, logger): Promise<void> {
+        try {
+            const queryBody = this.buildElementQuery(config, logger);
+            queryBody.sort = new ElasticSearchSortField('name', 'asc', '_last');
+
+            if (!config.startTime) {
+                config.startTime = moment().startOf('month').format();
+            }
+
+            if (!config.endTime) {
+                config.endTime = moment().format();
+            }
+
+            const metricRequest = this.buildRequest(config, queryBody, 'metrics/elasticsearch/metricQuery');
+            const metricResponse = await request(metricRequest);
+            if (config.excessivelogging) {
+                logger.info('METRIC REQUEST');
+                logger.info(JSON.stringify(metricRequest, null, 2));
+                logger.info('METRIC RESPONSE');
+                logger.info(JSON.stringify(metricResponse, null, 2));
+            }
+            const rows = [];
+
+            for (const el of metricResponse.page.content) {
```

Comment:
Can you use a full variable name like `element`? No reason to make the reader guess what `el` stands for.